### PR TITLE
Tinderbox platform fixes

### PR DESCRIPTION
**Pull request by sydvicious**
Supercedes #258 and #259 

- Fix --type=tinderbox --platform=win32 --branch=mozilla-beta --extension=txt (or anything else)
- Fix --type=tinderbox --platform=mac --branch=mozilla-beta
Added tests.

### CODE REVIEW

**Comment by sydvicious** [comment]
@whimboo Please review.

File: tests/tinderbox_scraper/test_tinderbox_scraper.py [hunk 1]
```diff
@@ -142,6 +142,22 @@
      'target': 'mozilla-central-firefox-25.0a1.en-US.linux-i686.txt',
      'target_url': 'firefox/tinderbox-builds/mozilla-central-linux/'
                    '1374583608/firefox-25.0a1.en-US.linux-i686.txt'},
+    # -a firefox -t tinderbox -p win32 --branch=mozilla-beta --extension=txt
```
Comment (whimboo) [inline tests/tinderbox_scraper/test_tinderbox_scraper.py]: `--extension=txt` is not only valid for the `mozilla-beta` branch but for all branches. Please check the test above. Given that all other tests are based on `mozilla-central` please convert the win32 and mac case to `mozilla-central`.

File: mozdownload/scraper.py [hunk 2]
```diff
@@ -917,7 +917,7 @@ def platform_regex(self):
 
         PLATFORM_FRAGMENTS = {'linux': 'linux',
                               'linux64': 'linux64',
-                              'mac': 'macosx',
+                              'mac': 'macosx64',
```
Comment (whimboo) [inline mozdownload/scraper.py]: This has already been fixed in PR #235. So the change should be obsolete now.

**Comment by whimboo** [comment]
Thanks @sydvicious for the PR. Please see my inline comments. Once addressed ask for review again, and I'm sure we can get it merged then.

**Comment by sydvicious** [comment]
Superceded by #264

**Pull request closed without merging**